src/devices/src/virtio/net/tap.rs

```diff
@@ -163,6 +166,40 @@ impl Tap {
 
         Ok(())
     }
+
+    pub fn if_index(&self) -> i32 {
+        let sock = create_socket();
+        let ifreq = IfReqBuilder::new()
+            .if_name(&self.if_name)
+            .execute(&sock, c_ulong::from(net_gen::sockios::SIOCGIFINDEX))
+            .unwrap();
+
+        unsafe { *ifreq.ifr_ifru.ifru_ivalue.as_ref() }
+    }
+
+    /// Enable the tap interface.
+    pub fn enable(&self) {
+        // Disable IPv6 router advertisment requests
+        Command::new("sh")
+            .arg("-c")
+            .arg(format!(
+                "echo 0 > /proc/sys/net/ipv6/conf/{}/accept_ra",
+                self.if_name_as_str()
+            ))
+            .output()
+            .unwrap();
+
+        let sock = create_socket();
+        IfReqBuilder::new()
+            .if_name(&self.if_name)
+            .flags(
+                (net_gen::net_device_flags_IFF_UP
+                    | net_gen::net_device_flags_IFF_RUNNING
+                    | net_gen::net_device_flags_IFF_NOARP) as i16,
+            )
+            .execute(&sock, c_ulong::from(net_gen::sockios::SIOCSIFFLAGS))
+            .unwrap();
+    }
```

Comment:
Not sure if straightforward but these functions should definitely go in `net::test_utils` somehow.
They definitely don't make sense as `firecracker` binary code.

Reply:
I have moved them in net/test_utils.rs. Take a look please.